SocialCareCaseViewerApi.Tests/V1/Gateways/MongoGatewayTests.cs

```diff
@@ -0,0 +1,244 @@
+            }
+            retrievedObject?.Count.Should().Be(1);
+            retrievedObject?[0].Should().BeEquivalentTo(_testObjectForMongo);
+
+            _mongoGateway.InsertRecord("test-collection-name", new TestObjectForMongo());
+            if (_testObjectForMongo != null)
+            {
+                retrievedObject = _mongoGateway.LoadRecords<TestObjectForMongo>("test-collection-name");
+            }
+            retrievedObject?.Count.Should().Be(2);
+        }
+
+        [Test]
+        public void CanLoadARecordByACustomProperty()
+        {
+            _mongoGateway.InsertRecord("test-collection-name", _testObjectForMongo);
+
+            TestObjectForMongo? retrievedObject = null;
+
+            if (_testObjectForMongo != null)
+            {
+                retrievedObject =
+                    _mongoGateway.LoadRecordByProperty<TestObjectForMongo, string>(
+                        "test-collection-name", "Property1", _testObjectForMongo.Property1);
+            }
+            retrievedObject.Should().BeEquivalentTo(_testObjectForMongo);
+        }
+
+        [Test]
+        public void LoadRecordByCustomPropertyReturnsNullWhenNoMatchFound()
+        {
+            TestObjectForMongo? retrievedObject = null;
+
+            if (_testObjectForMongo != null)
+            {
+                retrievedObject =
+                    _mongoGateway.LoadRecordByProperty<TestObjectForMongo, string>(
+                        "test-collection-name", "Property1", _testObjectForMongo.Property1);
+            }
+            retrievedObject.Should().BeNull();
+        }
+
+        [Test]
+        public void CanLoadMultipleRecordsByACustomProperty()
+        {
+            _mongoGateway.InsertRecord("test-collection-name", _testObjectForMongo);
+            if (_testObjectForMongo != null)
+            {
+                _mongoGateway.InsertRecord("test-collection-name", new TestObjectForMongo { Property1 = _testObjectForMongo.Property1 });
+            }
+
+            List<TestObjectForMongo>? retrievedObject = null;
+
+            if (_testObjectForMongo != null)
+            {
+                retrievedObject =
+                    _mongoGateway.LoadMultipleRecordsByProperty<TestObjectForMongo, string>(
+                        "test-collection-name", "Property1", _testObjectForMongo.Property1)
+                        .ToList();
+            }
+            retrievedObject?.Count.Should().Be(2);
+        }
+
+        [Test]
+        public void LoadMultipleRecordsByCustomerPropertyWithNoMatchReturnsEmptyList()
+        {
+            List<TestObjectForMongo>? retrievedObject = null;
+
+            if (_testObjectForMongo != null)
+            {
+                retrievedObject =
+                    _mongoGateway.LoadMultipleRecordsByProperty<TestObjectForMongo, string>(
+                            "test-collection-name", "Property1", _testObjectForMongo.Property1)
+                        .ToList();
```

Comment:
Do we need to do `ToList()` here? Looks like the `_mongoGateway.LoadMultipleRecordsByProperty` does it already.

Reply:
Good suggestion, I had to update the interface to say a List<> was being returned and not an IEnumerable<>.

I'm all about performance gains.